draft-irtf-cfrg-randomness-improvements.md

```diff
@@ -254,11 +256,17 @@ that generates private values, such as key shares, with:
 G'(n) = HKDF-Expand(HKDF-Extract(H(Sig(sk, tag1)), G(L)), tag2, n)
 ~~~
 
+# Implementation Guidance
+
+Recall that the wrapper defined in {{wrapper}} requires L >= n - L', where L is the Extract output
+length and n is the desired amount of randomness. Some applications may require n to exceed this bound.
+Wrapper implementations SHOULD support this use case by invoking G'(L) multiple times and concatenating
+the results.
```

Comment:
"by invoking G'(L) multiple times"  - but the construction is defined for G'(n).
I'd propose to write it as follows:
"Some applications may require more randomness than L+L'. Wrapper implementations SHOULD support this use case by invoking G'(n) multiple times and concatenating the results."

Reply:
Hmm, that's not quite right. Here's what the wrapper would do, omitting edge cases:

```
def G(n):
   bytes = []
   for i in range((n / L) + 1):
      bytes.append(G'(L))
   return bytes
```

Reply:
Yes, my "n" was misleading - I meant something like "invoking G'(*)" multiple times, removing mentioning L (because L and L' are defined parameters of any construction instance, while n is not). I agree that "G'(n)" would be confusing.

In your pseudocode you write "G'(L)", then in the paragraph you write "G'(L')". But the maximum possible amount of bytes is L+L', neither L or L'.

Hence we should write "Some applications may require more randomness than L+L'. Wrapper implementations SHOULD support this use case by invoking G'(L+L') multiple times and concatenating the results.", shouldn't we?

Reply:
I'm just going to remove the parameter entirely. Implementations could choose to call G' with L, L + L', or whatever else (within bounds) as they see fit.

Reply:
Yes, it seems the best option, great!
Thanks, Chris!

Reply:
No problem! 👍